src/commands/channel/promote.ts

```diff
@@ -10,58 +10,60 @@ import * as os from 'os';
 import { bold } from 'chalk';
 import { valid as validSemVer } from 'semver';
 import { ShellString } from 'shelljs';
+import { Interfaces } from '@oclif/core';
 
-import { FlagsConfig, SfdxCommand, flags } from '@salesforce/command';
-import { Messages, SfError } from '@salesforce/core';
-import { AnyJson, asArray, ensureArray, ensureNumber, ensureString } from '@salesforce/ts-types';
-
+import { Logger, Messages, SfError } from '@salesforce/core';
+import { AnyJson, ensureString } from '@salesforce/ts-types';
+import { SfCommand, Flags, arrayWithDeprecation } from '@salesforce/sf-plugins-core';
 import { AmazonS3 } from '../../amazonS3';
-import { Flags, verifyDependencies } from '../../dependencies';
+import { verifyDependencies } from '../../dependencies';
 import { PluginCommand } from '../../pluginCommand';
 import { CLI, Channel, S3Manifest, VersionShaContents } from '../../types';
 
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/plugin-release-management', 'channel.promote');
 const TARGETS = ['linux-x64', 'linux-arm', 'win32-x64', 'win32-x86', 'darwin-x64'];
-export default class Promote extends SfdxCommand {
+
+export default class Promote extends SfCommand<AnyJson> {
   public static readonly description = messages.getMessage('description');
+  public static readonly summary = messages.getMessage('summary');
   public static readonly examples = messages.getMessage('examples').split(os.EOL);
-  public static readonly flagsConfig: FlagsConfig = {
-    dryrun: flags.boolean({
+  public static flags = {
+    dryrun: Flags.boolean({
       char: 'd',
       default: false,
-      description: messages.getMessage('dryrun'),
+      summary: messages.getMessage('dryrun'),
     }),
-    target: flags.string({
+    'promote-to-channel': Flags.string({
       char: 't',
       default: Channel.STABLE,
-      description: messages.getMessage('target'),
+      summary: messages.getMessage('target'),
       // options: Object.values(Channel),
       required: true,
+      aliases: ['target'],
```

Comment:
Want to add `deprecateAliases: true` for the flags you've renamed?

Reply:
nah, I think it'll just create extra noise in the GHA logs.  

We can change our scripts if we ever really want to.